tests/common/test_additional_secrets_redaction.py

```diff
@@ -0,0 +1,345 @@
+"""Test customizing security admin segment traits."""
+
+import unittest
+from unittest.mock import Mock, patch
+
+import __init__
+
+import tests.common.test_common_constants as TestCommonConstants
+import tests.data_set.test_data_set_constants as TestDataSetConstants
+import tests.group.test_group_constants as TestGroupConstants
+import tests.resource.test_resource_constants as TestResourceConstants
+import tests.user.test_user_constants as TestUserConstants
+from pyracf import (
+    DataSetAdmin,
+    GroupAdmin,
+    ResourceAdmin,
+    SecurityRequestError,
+    UserAdmin,
+)
+
+# Resolves F401
+__init__
+
+
+class TestAdditionalSecretsRedaction(unittest.TestCase):
+    maxDiff = None
+
+    # ============================================================================
+    # User Administration
+    # ============================================================================
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_user_admin_custom_secret_redacted_on_success(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(additional_secret_traits=["omvs:uid"])
+        call_racf_mock.side_effect = [
+            TestUserConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestUserConstants.TEST_ALTER_USER_RESULT_EXTENDED_SUCCESS_XML,
+        ]
+        result = user_admin.alter(
+            "squidwrd",
+            traits=TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED,
+        )
+        self.assertEqual(
+            result,
+            TestCommonConstants.TEST_ALTER_USER_RESULT_SUCCESS_UID_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED["omvs:uid"],
+            result,
+        )
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_user_admin_custom_secret_redacted_on_error(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(additional_secret_traits=["omvs:uid"])
+        call_racf_mock.side_effect = [
+            TestUserConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestUserConstants.TEST_ALTER_USER_RESULT_ERROR_XML,
+        ]
+        with self.assertRaises(SecurityRequestError) as exception:
+            user_admin.alter(
+                "squidwrd",
+                traits=TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_UID_ERROR,
+            )
+        self.assertEqual(
+            exception.exception.result,
+            TestCommonConstants.TEST_ALTER_USER_RESULT_ERROR_UID_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_UID_ERROR["omvs:uid"],
+            exception.exception.result,
+        )
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_user_admin_custom_secret_redacted_when_complex_characters(
```

Comment:
Is there no failure scenario to test for installation data?

Reply:
I did not add one, but the redaction of secrets is functionally no different on success cases or error cases with the potential exception of messages concerning the field. I am not aware of any specific messages to INSTALLATION DATA. I can add more test cases when formally stabilizing Installation data.